src/problems/Albany_Layouts.cpp

```diff
@@ -201,14 +201,24 @@ Albany::Layouts::Layouts (int worksetSize, int numVertices, int numNodes, int nu
   useCollapsedSidesets = collapsed_sidesets;
 
   // Collapsed sideset layouts to ensure contiguous memory access for efficient GPU evaluation
+  // TODO: using meshspecs struct to set sideset workset size doesn't work for all cases,
+  //        so we are using the naive lower bound here until we can find a better bound
+  // sidesetWorksetSize = (sidesetWorksetSize > 0) ? sidesetWorksetSize : worksetSize*numSides;
+  sidesetWorksetSize = worksetSize*numSides;
```

Comment:
I was just thinking, that maybe we could switch to `PHX::Layout` instead of `PHX::MDALayout`. The former allows to defer the specification of the extents to a later time. For us, that'd be after the whole mesh has been created, so that we know the exact number of sides on the sideset, and can allocate smaller side layouts.

The main drawback is that with dynamic layouts, you _need_ to call `setFieldData` for each required/evaluated field in `postRegistrationSetup`, while now fields are allocated at construction time, so the `setFieldData` step is superfluous (and sometimes I don't even do it in the evaluators I write). Also, you would not be able to get the number of nodes or quad points from a layout at construction time inside the evaluators (like we sometimes do).

Just a thought, for later.

P.s: obviously, `PHX::Layout` still inherits from `PHX::DataLayout`, so the modifications would be minimal.